### ISSUE TITLE
MaskRCNN::set_trainable (probably) overwrites train_bn=False

### ISSUE THREAD

**Issue by syrix** [matterport/Mask_RCNN#400]
While constructing the model the batch-norm layers are set to be non-trainable when `train_bn = False`. The `train` function of the MaskRCNN class sets certain layers to trainable by calling `set_trainable`, including all batch-norm layers (if the regex is e.g. "*").

Due to this the batch-norm layers are (probably) always set to trainable if the train function is called with `layers = 'all'`.

To fix this the batch norm layers should be excluded from being affected by the set_trainable function, e.g. by changing the regex to always exclude certain names (even though that is a bit hacky...). 

Somewhat related the parameter in the Layer constructor called in line 77 of model.py should be `trainable` not `training`. This means that
`return super(self.__class__, self).call(inputs, training=training)` should be `return super(self.__class__, self).call(inputs, trainable=training)`
[See here](https://github.com/keras-team/keras/blob/04c11f8c13b1182b43eaed36dbfdedb23efd38b9/keras/engine/topology.py#L288). 
Until this is changed setting train_bn unfortunately does not do anything.

I unfortunately don't have access to a GPU for a few weeks, so I can not check if those two issues really exist, but if they do, they probably affects the performance of the model a fair bit.

**Comment by syrix** [comment]
My mistake. The training parameter is passed to the train-method, not to init => it works as it is.
